### PR TITLE
build: various improvements

### PR DESCRIPTION
**Pull request by tiagolobocastro**
ci(gha/submodule): ensure submodules point to the right branch
    
    Will help avoid mistakes if anyone uses update --remote...
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    chore: don't show crate version on log
    
    We never update the crate version, and it's actually very misleading...
    So let's hide it until we start to update it properly.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

---

    build(images): add tag argument
    
    Allows one to explicitly build the binaries with a given tag, as opposed to the alias-tag
    which simply adds a docker image tag alias.
    
    Signed-off-by: Tiago Castro <tiagolobocastro@gmail.com>

### CODE REVIEW

File: scripts/git/set-submodule-branches.sh [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    *)
+      echo "Unknown option: $1"
+      exit 1
+      ;;
+  esac
+done
+
+
+
+if [ "$BRANCH" == "develop" ]; then
```
Comment (niladrih) [inline scripts/git/set-submodule-branches.sh]: Could both of these be inside the same if condition. They both do the same thing `SET_BRANCH="${BRANCH}"`

**Review by niladrih — COMMENTED** [review]
minor comment

File: scripts/git/set-submodule-branches.sh (hunk 1 above)
Comment (tiagolobocastro) [reply scripts/git/set-submodule-branches.sh]: Yes indeed

**Review by niladrih — APPROVED** [review]

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
bors merge

**Comment by tiagolobocastro** [comment]
hmm strange errors from the bdd tests...
bors merge

**Comment by tiagolobocastro** [comment]
Caused by this: https://github.com/docker/docker-py/issues/3113
I've pinned requests for now.
bors merge

**Pull request merged**